api/v1/mongodbcommunity_types.go

```diff
@@ -119,6 +119,69 @@ type MongoDBCommunitySpec struct {
 	// Prometheus configurations.
 	// +optional
 	Prometheus *Prometheus `json:"prometheus,omitempty"`
+
+	// Additional options to be appended to the connection string. These options apply to the entire resource and to each user.
+	// +kubebuilder:validation:Type=object
+	// +optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +nullable
+	AdditionalConnectionStringConfig MapWrapper `json:"additionalConnectionStringConfig,omitempty"`
+}
+
+// Wrapper for a map to be used by other structs.
+// The CRD generator does not support map[string]interface{}
+// on the top level and hence we need to work around this with
+// a wrapping struct.
+type MapWrapper struct {
+	Object map[string]interface{} `json:"-"`
+}
+
+// MarshalJSON defers JSON encoding to the wrapped map
+func (m *MapWrapper) MarshalJSON() ([]byte, error) {
+	return json.Marshal(m.Object)
+}
+
+// UnmarshalJSON will decode the data into the wrapped map
+func (m *MapWrapper) UnmarshalJSON(data []byte) error {
+	if m.Object == nil {
+		m.Object = map[string]interface{}{}
```

Comment:
With this code, does it make sense to use `// +nullable` in the type definition?

Reply:
I am not completely sure that it's related. I believe that `// +nullable` allows the field `AdditionalConnectionStringConfig` itself to be **nil**, and I don't think it's related to the `Object` field of `MapWrapper`, which could be **nil** anyways. I may be wrong. However, I agree that removing the marker would be better.

Reply:
Upon further testing, I notice that having this marker allows for using the SetOption method without actually checking if the map is nil or not. This way, it helps to avoid any nil pointer exceptions. So having it does make sense.

Reply:
Thanks for checking!